yarn-project/archiver/src/archiver/data_retrieval.ts

```diff
@@ -62,12 +64,53 @@ export async function retrieveBlocks(
       break;
     }
 
-    const newBlocks = await processBlockLogs(publicClient, expectedNextL2BlockNum, l2BlockProcessedLogs);
-    retrievedBlocks.push(...newBlocks);
+    const newBlockMetadata = await processBlockLogs(publicClient, expectedNextL2BlockNum, l2BlockProcessedLogs);
+    retrievedBlockMetadata.push(...newBlockMetadata);
     searchStartBlock = l2BlockProcessedLogs[l2BlockProcessedLogs.length - 1].blockNumber! + 1n;
-    expectedNextL2BlockNum += BigInt(newBlocks.length);
+    expectedNextL2BlockNum += BigInt(newBlockMetadata.length);
   } while (blockUntilSynced && searchStartBlock <= searchEndBlock);
-  return { nextEthBlockNumber: searchStartBlock, retrievedData: retrievedBlocks };
+  return { nextEthBlockNumber: searchStartBlock, retrievedData: retrievedBlockMetadata };
+}
+
+/**
+ * Fetches new L2 Blocks.
+ * @param publicClient - The viem public client to use for transaction retrieval.
+ * @param rollupAddress - The address of the rollup contract.
+ * @param blockUntilSynced - If true, blocks until the archiver has fully synced.
+ * @param searchStartBlock - The block number to use for starting the search.
+ * @param searchEndBlock - The highest block number that we should search up to.
+ * @param expectedNextL2BlockNum - The next L2 block number that we expect to find.
+ * @returns An array of L2 Blocks and the next eth block to search from
+ */
+export async function retrieveBlockBodiesFromDataAvailability(
+  publicClient: PublicClient,
+  rollupAddress: EthAddress,
+  blockUntilSynced: boolean,
+  searchStartBlock: bigint,
+  searchEndBlock: bigint,
+  expectedNextL2BlockNum: bigint,
+): Promise<DataRetrieval<[Body, Buffer]>> {
+  const retrievedBlockBodies: [Body, Buffer][] = [];
+
+  do {
+    if (searchStartBlock > searchEndBlock) {
+      break;
+    }
+    const l2BlockProcessedLogs = await getL2BlockProcessedLogsFromDataAvailabilityOracle(
```

Comment:
There is no `L2BlockProcessed` event in availability oracle.

Reply:
The event the function is looking for is `TxsPublished` so would probably make sense to name it based on that as well 👍